File: app/src/main/java/eu/toma/dev/playground/warcraft/mounts/MountDb.java
package eu.toma.dev.playground.warcraft.mounts;

import java.util.List;

import javax.inject.Inject;

import eu.toma.dev.playground.warcraft.mounts.di.Mounts;
import io.reactivex.Observable;

@Mounts
public class MountDb
{
    private List<Mount> mounts;

    @Inject
    public MountDb()
    {
    }

    public Observable<List<Mount>> mountsFromDb()
    {
        if(mounts == null){
            return Observable.error(new Exception("db null mounts"));
        } else {
            return Observable.fromIterable(mounts).toList().toObservable();
        }
    }

    public void save(List<Mount> mounts)
    {
        this.mounts = mounts;
    }
}

File: app/src/main/kotlin/warcraft/mounts/MountDb.kt
package warcraft.mounts

import javax.inject.Inject

import io.reactivex.Observable

@warcraft.mounts.di.Mounts
class MountDb
@Inject
constructor() {
    private var mounts: List<Mount>? = null

    fun mountsFromDb(): Observable<List<Mount>> {
        if (mounts == null) {
            return Observable.error<List<Mount>>(Exception("db null mounts"))
        } else {
            return Observable.fromIterable(mounts).toList().toObservable()
        }
    }

    fun save(mounts: List<Mount>) {
        this.mounts = mounts
    }
}
